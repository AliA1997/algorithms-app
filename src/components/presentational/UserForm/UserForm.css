.form-wrapper {
    display: grid;
    grid-template-rows: 10% 20% 10% 10% 10% 10% 10% 10% 10%;
    width: 100%;
    min-height: 100%;
}
.form-wrapper > * {
    width: 100%;
    height: 100%;
    grid-row: auto;
}
.form-wrapper > *:nth-child(2) {
    margin-left: auto;
    margin-right: auto;
    background: red;
    height: 10em;
    width: 10em;
    border-radius: 50%;
}

.form-label {
    font-size: 1.2em;
    font-weight: 700;
    text-align: center;
    display: block;
}

.favorite-programming-languages-div {
    width: 100%;
    height: 80%;
    display: grid;
}
.programming-language-card {
    grid-row: auto;
    display: flex;
    border: solid 2px #000;
    border-top-left-radius: 100px;
    border-top-right-radius: 100px;
    border-bottom-left-radius: 100px;
    border-bottom-right-radius: 100px;
    justify-content: space-around;
    align-items: center;
    width: 100%;
    padding: 2.5%;
}
.programming-language-card > img {
    border-radius: 50%;
    width: 50%;
}
.programming-language-card > img[src] {
    height: 90%;
    padding: 10px;
}
.programming-language-card > p {
    font-size: 1.5em;
    font-weight: 700;
}