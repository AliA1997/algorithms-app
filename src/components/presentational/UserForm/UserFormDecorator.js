//import withHandler from recompose
import { withHandlers, compose } from 'recompose';
//import withRouter from react-router-dom 
import { withRouter } from 'react-router-dom';

const withHandlersDecorator = withHandlers({
    handleUsername: ({handleUsername}) => e => {
      console.log(e.target.value);
      handleUsername(e.target.value)  
    },
    handleName: ({handleName}) => e => {
        console.log('Hit-----------');
        handleName(e.target.value);
    },
    handleEmail: ({handleEmail}) => e => {
        console.log('Hit-----------');
        handleEmail(e.target.value);
    },
    handlePassword: ({handlePassword}) => e => {
        console.log('Hit-----------');
        handlePassword(e.target.value);
    },
    handlePhoneNumber: ({handlePhoneNumber}) => e => {
        console.log('Hit-----------');
        handlePhoneNumber(e.target.value);
    },
    handleExperience: ({handleExperience}) => e => {
        console.log('Hit-----------');
        handleExperience(e.target.value);
    },
    handleEdu: ({handleEdu}) => e => {
        console.log('Hit-----------');
        handleEdu(e.target.value);
    },
    onClick: ({register, history}) => () => {
        register();
        history.push('/');
    }
})

// export default withHandleDecorator;
export default withHandlersDecorator;