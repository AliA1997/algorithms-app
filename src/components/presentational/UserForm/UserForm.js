import React from 'react';
import uuid from 'uuid';
//import Suggestions Component.
import Suggestions from '../Suggestions/Suggestions';
//import data needed for the programming languages 
import programmingLanguages from '../../../data/programmingLanguages.json';
// import TiPlus from 'react-icons/lib/ti/plus';
import { withRouter } from 'react-router-dom';
//import decorator 
import decorator from './UserFormDecorator';
//import css file for styling
import './UserForm.css';

export const UserForm = (props) => {
    console.log(props.favorite_programming_languages);
    return (
        <div className='form container'>
            <div className='form-wrapper'>
                <h1 className='header'>Register</h1>
                <img src={props.profile_picture || `https://robohash.org/${uuid.v4()}`} value={props.profile_picture} />
                <div>
                    <label className='form-label'>Profile Picture</label>
                    <input type='file' onChange={props.upload} />
                </div>
                <div>
                    <label className='form-label'>Username</label>
                    <input type='text' onChange={props.handleUsername} value={props.username} />
                </div>
                <div>
                    <label className='form-label'>Name</label>
                    <input type='text' onChange={props.handleName} value={props.name} />
                </div>
                <div>
                    <label className='form-label'>Email</label>
                    <input type='text' onChange={props.handleEmail} value={props.email} />
                </div>
                <div>
                    <label className='form-label'>Password</label>
                    <input type='password' onChange={props.handlePassword} value={props.password} />
                </div>
                <div>
                    <label className='form-label'>Programming Experience</label>
                    <input type='text' onChange={props.handleExperience} value={props.experience} />
                </div>
                <div>
                    <label className='form-label'>Phone Number</label>
                    <input type='number' onChange={props.handlePhoneNumber} value={props.phone_number} />
                </div>
                <div>
                    <label className='form-label'>Education Completed</label>
                    <select onChange={props.handleEdu} value={props.recent_edu_completed}>
                        <option defaultValue=''></option>
                        {props.edu.map((el, i) => <option key={i} value={el}>{el}</option>)}
                    </select>
                </div>
                <div>
                    <label className='form-label'>Favorite Programming Languages</label>
                    <input list='programming-languages' onChange={(e) =>  props.handleCurrentProgrammingLanguage(e.target.value)} 
                    value={props.currentProgrammingLanguage} />
                    <Suggestions suggestions={programmingLanguages} clickSuggestion={props.handleCurrentProgrammingLanguage}
                    suggestion={props.currentProgrammingLanguage} showSuggestion={props.currentProgrammingLanguage ? true : false}/>
                    <button className='add-button' onClick={props.addToFavProgrammingLanguages}>+</button>
                </div>
                <div className='favorite-programming-languages-div'>
                    {props.favorite_programming_languages.length ?
                     props.favorite_programming_languages.map((favLang, i) => { 
                     return <div className='programming-language-card' key={i} onClick={e => props.removeFromFavProgrammingLanguages(e, favLang.name.toLowerCase())}>
                                <p>{favLang.name}</p>
                                <img src={favLang.image} alt={favLang.name} />
                            </div>
                     }) : null}
                </div>
                <button className='btn' onClick={props.onClick}>Register</button>
            </div>
        </div>
    );
};


const UserFormDecorated = decorator(UserForm);
export default withRouter(UserFormDecorated);