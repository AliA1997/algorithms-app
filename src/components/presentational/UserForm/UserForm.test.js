//make sure to import react
import React from 'react';
//import the renderer for create a mock component with the component as a argument 
import render from 'react-test-renderer';
import { UserForm } from './UserForm';

const edu = [
    'Some High School',
    'High School or GED',
    'Some College',
    'AS',
    'BS',
    'MS',
    'PHD'
]
describe('Check if the component renders correctly', () => {
    test('Test if component matches snapshot', () => {
        const tree = render.create(<UserForm edu={edu}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})