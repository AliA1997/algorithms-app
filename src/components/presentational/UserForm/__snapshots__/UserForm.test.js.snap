// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="form container"
>
  <div
    className="form-wrapper"
  >
    <h1
      className="header"
    >
      Register
    </h1>
    <div>
      <label
        className="form-label"
      >
        Username
      </label>
      <input
        onChange={undefined}
        type="text"
        value={undefined}
      />
    </div>
    <div>
      <label
        className="form-label"
      >
        Name
      </label>
      <input
        onChange={undefined}
        type="text"
        value={undefined}
      />
    </div>
    <div>
      <label
        className="form-label"
      >
        Email
      </label>
      <input
        onChange={undefined}
        type="text"
        value={undefined}
      />
    </div>
    <div>
      <label
        className="form-label"
      >
        Password
      </label>
      <input
        onChange={undefined}
        type="password"
        value={undefined}
      />
    </div>
    <div>
      <label
        className="form-label"
      >
        Programming Experience
      </label>
      <input
        onChange={undefined}
        type="text"
        value={undefined}
      />
    </div>
    <div>
      <label
        className="form-label"
      >
        Phone Number
      </label>
      <input
        onChange={undefined}
        type="number"
        value={undefined}
      />
    </div>
    <div>
      <label
        className="form-label"
      >
        Education Completed
      </label>
      <select
        onChange={undefined}
        value={undefined}
      >
        <option
          selected={true}
          value=""
        />
        <option
          value="Some High School"
        >
          Some High School
        </option>
        <option
          value="High School or GED"
        >
          High School or GED
        </option>
        <option
          value="Some College"
        >
          Some College
        </option>
        <option
          value="AS"
        >
          AS
        </option>
        <option
          value="BS"
        >
          BS
        </option>
        <option
          value="MS"
        >
          MS
        </option>
        <option
          value="PHD"
        >
          PHD
        </option>
      </select>
    </div>
    <button
      className="btn"
      onClick={undefined}
    >
      Register
    </button>
  </div>
</div>
`;

exports[`Check if the component renders correctly Test if component matches snapshot 1`] = `
<div
  className="form container"
>
  <div
    className="form-wrapper"
  >
    <h1
      className="header"
    >
      Register
    </h1>
    <div>
      <label
        className="form-label"
      >
        Username
      </label>
      <input
        onChange={undefined}
        type="text"
        value={undefined}
      />
    </div>
    <div>
      <label
        className="form-label"
      >
        Name
      </label>
      <input
        onChange={undefined}
        type="text"
        value={undefined}
      />
    </div>
    <div>
      <label
        className="form-label"
      >
        Email
      </label>
      <input
        onChange={undefined}
        type="text"
        value={undefined}
      />
    </div>
    <div>
      <label
        className="form-label"
      >
        Password
      </label>
      <input
        onChange={undefined}
        type="password"
        value={undefined}
      />
    </div>
    <div>
      <label
        className="form-label"
      >
        Programming Experience
      </label>
      <input
        onChange={undefined}
        type="text"
        value={undefined}
      />
    </div>
    <div>
      <label
        className="form-label"
      >
        Phone Number
      </label>
      <input
        onChange={undefined}
        type="number"
        value={undefined}
      />
    </div>
    <div>
      <label
        className="form-label"
      >
        Education Completed
      </label>
      <select
        onChange={undefined}
        value={undefined}
      >
        <option
          selected={true}
          value=""
        />
        <option
          value="Some High School"
        >
          Some High School
        </option>
        <option
          value="High School or GED"
        >
          High School or GED
        </option>
        <option
          value="Some College"
        >
          Some College
        </option>
        <option
          value="AS"
        >
          AS
        </option>
        <option
          value="BS"
        >
          BS
        </option>
        <option
          value="MS"
        >
          MS
        </option>
        <option
          value="PHD"
        >
          PHD
        </option>
      </select>
    </div>
    <button
      className="btn"
      onClick={undefined}
    >
      Register
    </button>
  </div>
</div>
`;
